zkevm-circuits/src/evm_circuit/util/math_gadget/modulo.rs

```diff
@@ -0,0 +1,212 @@
+use crate::{
+    evm_circuit::util::{
+        self, constraint_builder::ConstraintBuilder, math_gadget::*, sum, CachedRegion,
+    },
+    util::Expr,
+};
+use eth_types::{Field, ToLittleEndian, Word};
+use halo2_proofs::plonk::Error;
+
+/// Constraints the words a, n, r such that:
```

Comment:
"Constrain the words a, n, r such that:" or "Constraints for the words a, n, r:"

Reply:
fixed